src/Runner/Runner.php

```diff
@@ -297,4 +461,31 @@ private function dispatchEvent(string $name, Event $event): void
 
         $this->eventDispatcher->dispatch($event, $name);
     }
+
+    private function getFileIterator(): LintingResultAwareFileIteratorInterface
+    {
+        if (null === $this->fileIterator) {
+            throw new \RuntimeException('File iterator is not configured. Pass paths during Runner initialisation or set them after with `setFileIterator()`.');
+        }
+
+        $fileIterator = new \ArrayIterator(
```

Comment:
we have similar logic in src/Console/ConfigurationResolver.php , we shall need it only once.

Reply:
No, it's not the same logic, and this particular piece of code [was already in `Runner::fix()`](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/62280c4d4db85fb42a9e14e54d4a4d16e7751b71/src/Runner/Runner.php#L101-L111), it was just extracted to `getFileIterator()` in order to re-create iterator for each chunk (related to [this](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7777#discussion_r1470386544)).